Add browser location bar #36872

lanckmann opened the pull request:
### **Add location bar to Browser Panel #36872**

- Add location bar widget with navigation and copy functionality
- Users can paste/type paths to navigate directly to directories
- Copy button copies selected item's path to clipboard
- Location bar automatically updates to show selected item's path
- Includes Go button and keyboard shortcut (Enter) for navigation
- Fixes #36872

lanckmann committed:
Add location bar to Browser Panel

- Add location bar widget with navigation and copy functionality
- Users can paste/type paths to navigate directly to directories
- Copy button copies selected item's path to clipboard
- Location bar automatically updates to show selected item's path
- Includes Go button and keyboard shortcut (Enter) for navigation
- Resolves feature request #36872

lanckmann committed:
add QgsMessageBar include

nyalldawson commented:
@lanckmann can you add some screenshots please?

lanckmann committed:
Fix crash issues in browser location bar

lanckmann committed:
Fix browser crash when clicking folders- Fix copySelectedPath() to map proxy model index to source model- Fix updateLocationBar() to map proxy model index to source model

lanckmann committed:
batch hierarchy loading, caching, controlled events & auto‑refresh

baswein commented:
Thanks @lanckmann - That fixed the crash on click on folder name. 
I am now getting a crash when I try to open my Desktop folder. When I restart QGIS it says **WARNING    Plugin MetaSearch : This plugin is disabled because it previously crashed QGIS** . However I looked and it seemed like the plugin was not enabled when I started so i'm not sure what is going on there. Most other folders seem to work but some folders spin forever. Like my downloads folder. Maybe those with large files?

I also am unable to paste a path in and then navigate to it. Here are my error messages: 
2025-07-30T14:51:24     SUCCESS    Copy Path : Path copied to clipboard: C:/Users/bas/Documents/Adobe
2025-07-30T14:51:36     WARNING    Navigate to Path : Could not navigate to path: C:/Users/bas/Documents/Adobe

baswein commented:
I dug around a bit more and it seems like it is not an issue with the desktop but a folder on the desktop that has .adf files in it. 
Even if I create a dummy .adf file from a text file it crashes.

lanckmann committed:
Fix browser location bar navigation for paths(forward/backslash)

baswein commented:
Thanks @agiudiceandrea! 
So with your update the warning changed. It now knows it exists but can't figure out how to get there. 
**2025-07-31T09:45:44     SUCCESS    Copy Path : Path copied to clipboard: C:/Users/bas/Desktop/2
2025-07-31T09:45:53     WARNING    Navigate to Path : Could not navigate to path: C:/Users/bas/Desktop/2. The path exists but may not be accessible through the browser.**
Are you programing on a windows machine?

lanckmann committed:
browser location bar with cross-platform

baswein commented:
Still not working on the latest. 
I copy path from windows explorer: C:\Users\bas\Desktop\2
I paste it into the location bar.  
I get this error: **2025-08-01T11:11:40     WARNING    Navigate to Path : Could not navigate to path: C:/Users/bas/Desktop/2. The path exists but may not be accessible through the browser.**
In the error message the slashes are reversed. 
Although the location bar still looks like this 
<img width="301" height="30" alt="image" src="https://github.com/user-attachments/assets/c372a30c-aa06-4d57-9b3f-7782176f5d86" />
If I were to navigate to the folder in the qgis browser the path would be forward slashes. 
<img width="232" height="30" alt="image" src="https://github.com/user-attachments/assets/243a811e-ea89-495f-87d4-72e07b2b0f54" />
If I copy the path using the path using the location bar copy  and then paste I also get an error. 
**2025-08-01T11:30:34     SUCCESS    Copy Path : Path copied to clipboard: C:/Users/bas/Desktop/2
2025-08-01T11:30:48     WARNING    Navigate to Path : Could not navigate to path: C:/Users/bas/Downloads. The path exists but may not be accessible through the browser.**

Windows is fine with me pasting the forward slash version into the windows explorer but probably it should be copied and displayed using the conventions of the current OS.

lanckmann committed:
browser location bar update link

lanckmann committed:
Fix browser navigation path

lanckmann committed:
browser location bar fix slash

lanckmann committed:
Fix Windows path separator display in browser location

lanckmann committed:
last error

baswein commented:
The paths look good to me now! 
Still can't paste and navigate to a folder. 
**2025-08-06T09:40:50     SUCCESS    Copy Path : Path copied to clipboard: C:\tmp
2025-08-06T09:41:00     WARNING    Navigate to Path : Could not navigate to path: C:\tmp. The path exists but may not be accessible through the browser.**

lanckmann committed:
children bug path

lanckmann committed:
partial loading

lanckmann committed:
fix partial function

lanckmann committed:
last commit

lanckmann committed:
Fix browser location bar build error and simplify navigation logic

lanckmann committed:
Fix navigation issues where paths weren't found in browser model

baswein commented:
I think a icon based on expand all would be better than the group one. I took a crack at it here:
<img width="36" height="37" alt="image" src="https://github.com/user-attachments/assets/81e409ca-0b66-4d1a-9ec9-9b35a85d50e6" />
![mActionExpandTreeTo](https://github.com/user-attachments/assets/1e9604c6-2dc0-4a23-b104-65fce3145313)

baswein commented:
Or maybe this: 
<img width="54" height="51" alt="image" src="https://github.com/user-attachments/assets/50169ad6-5e06-4
![mActionExpandTreeTo_v2](https://github.com/user-attachments/assets/a3084dde-9227-4ca5-be66-0480fcde3147)
d89-9a99-733420b91d3f" />

lanckmann committed:
Fix crash when expanding unzipped folder

lanckmann committed:
Improve browser location bar path validation and case sensitivity

lanckmann committed:
Fix QgsMessageBar::pushWarning calls in browser widget

lanckmann committed:
Fix browser location bar navigation with case insensitive path resolution

lanckmann committed:
Use static QFileInfo::exists() for better performanceReplace QFileInfo(linkTarget).exists() with QFileInfo::exists(linkTarget)

lanckmann committed:
Trigger CI rebuild

lanckmann committed:
Apply black code formatting to Python files

Run black formatter on Python test files and scripts to meet
pre-commit hook requirements.

lanckmann committed:
Fix case-insensitive path resolution to correct all path

lanckmann committed:
Apply clang-format fixes to browser location bar

lanckmann committed:
Fix StackOverflow reference link

Update reference to correct answer: https://stackoverflow.com/a/77954785

lanckmann committed:
Apply clang-format fixes

lanckmann committed:
Refine browser location bar UI and fix non existent drive freeze

lanckmann committed:
Fix infinite recursion when navigating to drive root in location bar

jfbourdon commented:
With these last changes, it now fully addresses the related issue, for me at least. Any thoughts @baswein ?

lanckmann committed:
Remove unused browser copy path

nyalldawson commented on scripts/qgis_fixes/fix_pyqt.py:
```diff
@@ -1,5 +1,4 @@
-"""Migrate imports of PyQt4 to PyQt wrapper
-"""
+"""Migrate imports of PyQt4 to PyQt wrapper"""
```
There's a lot of unrelated changes here, can you please revert all these?

lanckmann committed:
Refactor browser location bar to simple Open Path button

lanckmann committed:
Merge remote-tracking branch 'origin/master' into add-browser-location-bar-36872

lanckmann committed:
Trigger CI

lanckmann committed:
Merge branch 'master' into add-browser-location-bar-36872

lanckmann committed:
Add both Open Path button and location bar for navigation.
Both implementations are included for testing. Please vote on which one to keep.

lanckmann commented:
I add both Open Path button and location bar for navigation. Both implementations
 are included for testing. Please vote on which one to keep.

lanckmann commented:
no stale
